web/libs/core/src/lib/Tour/Tour.tsx

```diff
@@ -0,0 +1,128 @@
+   *   Available actions:
+   *   - ACTIONS.CLOSE: User closed the tour
+   *   - ACTIONS.NEXT: User clicked next
+   *   - ACTIONS.PREV: User clicked back
+   *   - ACTIONS.RESET: Tour was reset
+   *   - ACTIONS.SKIP: User skipped the tour
+   *   - ACTIONS.START: Tour started
+   *   - ACTIONS.STOP: Tour stopped
+   * @param {number} data.index Current step index
+   * @param {string} data.type Event type
+   *   Available events:
+   *   - EVENTS.STEP_AFTER: After a step is completed
+   *   - EVENTS.STEP_BEFORE: Before a step starts
+   *   - EVENTS.TARGET_NOT_FOUND: Step target element not found
+   *   - EVENTS.TOUR_START: Tour started
+   *   - EVENTS.TOUR_END: Tour ended
+   * @param {string} data.status Tour status
+   *   Available statuses:
+   *   - STATUS.IDLE: Tour is idle/not started
+   *   - STATUS.RUNNING: Tour is running
+   *   - STATUS.PAUSED: Tour is paused
+   *   - STATUS.SKIPPED: Tour was skipped
+   *   - STATUS.FINISHED: Tour completed normally
+   *   - STATUS.ERROR: Tour encountered an error
+   *
+   * This handler manages:
+   * - Tour completion (close/skip/finish) by marking it viewed and stopping
+   * - Step navigation (next/prev) by updating the step index
+   *
+   * Can be extended to support:
+   * - Conditional step logic based on user interactions
+   * - Saving progress/state between sessions
+   * - Custom analytics tracking for each step
+   * - Dynamic step content based on application state
+   */
+  const handleTourCallback = useCallback(
+    (data: {
+      action: string;
+      index: number;
+      type: string;
+      status: string;
+    }) => {
+      const { action, index, type, status } = data;
+
+      // tour ends when
+      const shouldEndTour =
+        (status === STATUS.SKIPPED && state.run) || action === ACTIONS.CLOSE || status === STATUS.FINISHED;
+
+      if (shouldEndTour) {
+        // mark tour as viewed and update onboarding state if it's the final step or the tour was skipped
+        if (status === STATUS.SKIPPED || status === STATUS.FINISHED) {
+          tourContext?.setTourViewed(name, status === STATUS.SKIPPED, { index, action, type, status });
+        }
+        dispatch({ type: "STOP" });
+        return;
+      }
+
+      const isStepChange = type === EVENTS.STEP_AFTER || type === EVENTS.TARGET_NOT_FOUND;
+      if (isStepChange) {
+        const nextIndex = index + (action === ACTIONS.PREV ? -1 : 1);
+        dispatch({
+          type: "GOTO",
+          payload: { stepIndex: nextIndex },
+        });
+      }
+    },
+    [name, state.run],
+  );
+
+  return state.steps.length > 0 ? (
+    <JoyRide
+      {...state}
+      {...props}
+      callback={handleTourCallback}
+      styles={{
+        tooltip: {
+          width: "468px",
+        },
+        options: {
+          primaryColor: "var(--primary_link)",
+          textColor: "var(--sand_900)",
+          overlayColor: "rgba(0, 0, 0, 0.5)",
```

Comment:
can we define the width and the overlayColor using a css variable instead of hard setting them here? this will make future theming and white labeling efforts easier

Reply:
Looks to be no: https://github.com/gilbarbara/react-joyride/blob/main/src/styles.ts#L35

Reply:
Only takes in these style props, doesn't appear to have that option unfortunately.

Reply:
We'll have to dig more into that 🙁